apps/api/src/app/app.module.ts

```diff
@@ -44,6 +44,157 @@ const autoSchemaFile = environment.production
   ? true
   : 'apps/api/src/api.graphql'
 
+// asynchronously registered modules
+const nationalRegistryModule = NationalRegistryModule.register({
+  config: {
+    baseSoapUrl: environment.nationalRegistry.baseSoapUrl,
+    user: environment.nationalRegistry.user,
+    password: environment.nationalRegistry.password,
+    host: environment.nationalRegistry.host,
+  },
+})
+const nationalRegistryXRoadModule = NationalRegistryXRoadModule.register({
+  xRoadBasePathWithEnv: environment.nationalRegistryXRoad.url,
+  xRoadTjodskraMemberCode: environment.nationalRegistryXRoad.memberCode,
+  xRoadTjodskraApiPath: environment.nationalRegistryXRoad.apiPath,
+  xRoadClientId: environment.nationalRegistryXRoad.clientId,
+})
+const identityModule = IdentityModule.register({
+  modules: [nationalRegistryModule],
```

Comment:
Note the changes here.
Just pass in the registered module instead of passing down the environment variables and reregister it in the dependant module:
```typescript
IdentityModule.register({
  nationalRegistry: {
    baseSoapUrl: environment.nationalRegistry.baseSoapUrl,
    user: environment.nationalRegistry.user,
    password: environment.nationalRegistry.password,
    host: environment.nationalRegistry.host,
  },
}),
```